src/coreclr/jit/gentree.cpp

```diff
@@ -23482,6 +23509,15 @@ GenTree* Compiler::gtNewSimdShuffleNode(
             // swap the operands to match the encoding requirements
             retNode = gtNewSimdHWIntrinsicNode(type, op2, op1, NI_AVX512BW_PermuteVar32x16, simdBaseJitType, simdSize);
         }
+        else if (elementSize == 1)
```

Comment:
There's potentially a missing "todo" here to handle this using `pshufb` if we don't cross lanes and `VBMI` is not available.

I don't think it needs to be handled as part of this PR (or even for .NET 8 necessarily) since that scenario should only be limited to pre-cannon lake (and therefore first generation AVX-512 hardware). But, since it should just be refactoring the logic under the `if (simdSize == 32)` path to be shared; its probably worth having still.